sklearn/ensemble/_gb_losses.py

```diff
@@ -0,0 +1,881 @@
+        """Return the initial raw predictions.
+
+        Parameters
+        ----------
+        X : 2d array, shape (n_samples, n_features)
+            The data array.
+        estimator : estimator instance
+            The estimator to use to compute the predictions.
+
+        Returns
+        -------
+        raw_predictions : 2d array, shape (n_samples, K)
+            The initial raw predictions. K is equal to 1 for binary
+            classification and regression, and equal to the number of classes
+            for multiclass classification. ``raw_predictions`` is casted
+            into float64.
+        """
+        pass
+
+
+class RegressionLossFunction(LossFunction, metaclass=ABCMeta):
+    """Base class for regression loss functions.
+
+    Parameters
+    ----------
+    n_classes : int
+        Number of classes.
+    """
+    def __init__(self, n_classes):
+        if n_classes != 1:
+            raise ValueError("``n_classes`` must be 1 for regression but "
+                             "was %r" % n_classes)
+        super().__init__(n_classes)
+
+    def check_init_estimator(self, estimator):
+        """Make sure estimator has the required fit and predict methods.
+
+        Parameters
+        ----------
+        estimator : estimator instance
+            The init estimator to check.
+        """
+        if not (hasattr(estimator, 'fit') and hasattr(estimator, 'predict')):
+            raise ValueError(
+                "The init parameter must be a valid estimator and "
+                "support both fit and predict."
+            )
+
+    def get_init_raw_predictions(self, X, estimator):
+        return estimator.predict(X).reshape(-1, 1).astype(np.float64)
+
+
+class LeastSquaresError(RegressionLossFunction):
+    """Loss function for least squares (LS) estimation.
+    Terminal regions do not need to be updated for least squares.
+
+    Parameters
+    ----------
+    n_classes : int
+        Number of classes.
+    """
+
+    def init_estimator(self):
+        return DummyRegressor(strategy='mean')
+
+    def __call__(self, y, raw_predictions, sample_weight=None):
+        """Compute the least squares loss.
+
+        Parameters
+        ----------
+        y : 1d array, shape (n_samples,)
+            True labels.
+
+        raw_predictions : 2d array, shape (n_samples, K)
+            The raw_predictions (i.e. values from the tree leaves).
+
+        sample_weight : 1d array, shape (n_samples,), optional
+            Sample weights.
+        """
+        if sample_weight is None:
+            return np.mean((y - raw_predictions.ravel()) ** 2)
+        else:
+            return (1 / sample_weight.sum() * np.sum(
+                sample_weight * ((y - raw_predictions.ravel()) ** 2)))
+
+    def negative_gradient(self, y, raw_predictions, **kargs):
+        """Compute the negative gradient.
+
+        Parameters
+        ----------
+        y : 1d array, shape (n_samples,)
```

Comment:
@jnothman I went through the discussion in #13000 and other related PR/issues. I am bit behind the topic with my inactivity in the past months. Shall we use the single backtick to refer the glossary documentation. It does not matter here since we are dealing with private classes but for future review it will be useful.

Reply:
I wouldn't force the use of backticks where it's unlikely to help the user. So shape (n_samples,) I'd say we should avoid since it only reduces readability. But in general, yes, we can use backticks to link to glossary when there is no local method by the same name